tensorflow/python/ops/nn_ops.py

```diff
@@ -588,6 +610,17 @@ def __init__(self,
     self.op = build_op(num_spatial_dims, "VALID")
     self.call = self._with_space_to_batch_call
 
+    # We check if the direct dilated convolution can be used. If yes, we will
+    # replace the self.op with build_op_dilation. We only support direct cuDNN
+    # dilated convolution for 1D and 2D conv.
+    can_use_fused = False if build_op_dilation is None else True
```

Comment:
You can use num_spatial_dims (always set to a non-None numerical value above) in place of conv_dims here and simplify the code.